lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs

```diff
@@ -447,10 +465,204 @@ fileModeSpec =  do
+                                -- Assume the slotnumbers are very low
+                                , blockHeight = Quantity 0
+                                , headerHash = h
+                                , parentHeaderHash = hashA
+                                })
+                                mockTxs
+                                []
+                        let (FilteredBlock _ txs, cpB) = applyBlock fakeBlock cpA
+                        atomically $ do
+                            unsafeRunExceptT $ putCheckpoint testPk cpB
+                            unsafeRunExceptT $ putTxHistory testPk txs
+
+                let mockApplyBlock1 = mockApply (blockHash "block1")
+                            [ Tx (blockHash "tx1")
+                                [(TxIn (blockHash "faucet") 0, Coin 5)]
+                                [ TxOut (head ourAddrs) (Coin 5)
+                                ]
+                            ]
+                mockApplyBlock1
+                getAvailableBalance db `shouldReturn` 5
+
+                mockApply (blockHash "block2a")
+                            [ Tx
+                                (blockHash "tx2a")
+                                [ (TxIn (blockHash "tx1") 0, Coin 5)
+                                ]
+                                [ TxOut (dummyAddr "faucetAddr2") (Coin 3)
+                                , TxOut (ourAddrs !! 1) (Coin 2)
+                                ]
+                            ]
+
+                getAvailableBalance db `shouldReturn` 2
+                getTotalBalance db `shouldReturn` 2
+                getTxsInLedger db `shouldReturn` [(Outgoing, 3), (Incoming, 5)]
+                getPending db `shouldReturn` []
+
+
+
+                atomically $ void $ unsafeRunExceptT $ rollbackTo testPk (SlotId 0 0)
+                getTxsInLedger db `shouldReturn` []
+                -- getPending `shouldReturn` []
+                -- but got: [(Outgoing, 3)]
+
+                getAvailableBalance db `shouldReturn` 0
+                -- getTotalBalance `shouldReturn` 0
+                -- but got: 2
+
+                mockApplyBlock1
+
+                getTxsInLedger db `shouldReturn`[(Incoming, 5)]
+                getAvailableBalance db `shouldReturn` 0
+                getTotalBalance db `shouldReturn` 2
+
+                mockApply (blockHash "block2a")
+                            [ Tx
+                                (blockHash "tx2b")
+                                [ (TxIn (blockHash "tx1") 0, Coin 5)
+                                ]
+                                [ TxOut (ourAddrs !! 3) (Coin 3)
+                                , TxOut (ourAddrs !! 2) (Coin 2)
+                                ]
+                            ]
+                -- THESE ARE NOT EXPECTED, BUT PASS!!
+                -- getAvailableBalance `shouldReturn` 5
+                -- getTotalBalance `shouldReturn` 7
```

Comment:
I don't quite see why they "are not expected" though. This is 100% expected, but I agree, maybe not the best behavior we would like here. But, the code is simply doing doing what's it told to do.